### PR TITLE
Fixes errors when compiling with recent JDKS

### PR DESCRIPTION
**Pull request by EnricoScantamburlo**
Maven 3.0.5 does not work because it fails to dowload thru HTTP when
HTTPS is now required
Added a forced cast to Classloader to avoid ambigous method call and
made it compile

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -41,7 +41,7 @@ under the License.
         <url>https://issues.apache.org/jira/browse/NETBEANSINFRA</url>
     </issueManagement>
     <prerequisites>
-        <maven>3.0.5</maven>
+        <maven>3.6.3</maven>
```
Comment (matthiasblaesing) [inline pom.xml:44]: Is 3.6.3 the first maven release, that defaults to https as download url from maven central?

Comment (EnricoScantamburlo) [reply pom.xml:44]: I do not work, I think it's the latest, it is the one I tested and it worked, so I think it is a good value.

In NetBeans you can ignore that version number, if I remember correctly.

Comment (EnricoScantamburlo) [reply pom.xml:44]: 3.2.3 seems to be the first version that uses HTTPS

Comment (matthiasblaesing) [reply pom.xml:44]: Ok - I reread what prerequisites does and was reminded, that if you really want to enforce something maven-enforcer is the the right thing to use: "It is checked with Maven 2, it is not any more in Maven 3.". Lets keep it as is.

**Comment by matthiasblaesing** [comment]
Thank you.

**Pull request merged**